File: 1. Beginner/URI1047.py
a, b, c,d = input().split(" ")

HI = int(a)
MI = int(b)
HF = int(c)
MF = int(d)

totalH = HF - HI

if totalH < 0:
    totalH = 24 + (HF - HI)

totalM = MF - MI

if totalM < 0:
    totalM = 60 + (MF - MI)
    totalH-=1

if HI == HF and MI == MF:
    print("O JOGO DUROU 24 HORA(S) E 0 MINUTO(S)")
else:
    print("O JOGO DUROU %d HORA(S) E %d MINUTO(S)" %(totalH,totalM))

File: 1. Beginner/URI1047.cpp
#include <iostream>
#include <cstdio>

using namespace std;

int main() {
    int HI = 0, HF = 0, MI = 0, MF = 0, totalH = 0, totalM = 0;

    cin >> HI >> MI >> HF >> MF;

    totalH = HF - HI;

    if (totalH < 0) {
        totalH = 24 + (HF - HI);
    }

    totalM = MF - MI;
    if (totalM < 0) {
        totalM = 60 + (MF - MI);
        totalH--;
    }

    if (HI == HF && MI == MF) {
        printf("O JOGO DUROU 24 HORA(S) E 0 MINUTO(S)\n");
    } else {
        printf("O JOGO DUROU %d HORA(S) E %d MINUTO(S)\n", totalH, totalM);
    }
    return 0;
}
